tools/autograd/templates/python_variable_methods.cpp

```diff
@@ -206,6 +206,40 @@ static PyObject * THPVariable_detach_(PyObject* self, PyObject* args)
   END_HANDLE_TH_ERRORS
 }
 
+static double dispatch_to_CDouble(const Tensor & self) {
+  AutoNoGIL no_gil;
+  AutoGPU auto_gpu(self);
+  return self.toCDouble();
+}
+
+static int64_t dispatch_to_CLong(const Tensor & self) {
+  AutoNoGIL no_gil;
+  AutoGPU auto_gpu(self);
+  return self.toCLong();
+}
+
+static PyObject * THPVariable_float_scalar(PyObject* self, PyObject* args) {
+  HANDLE_TH_ERRORS
+  auto& self_ = reinterpret_cast<THPVariable*>(self)->cdata;
+  return wrap(dispatch_to_CDouble(self_));
+  END_HANDLE_TH_ERRORS
+}
+
+static PyObject * THPVariable_integral_scalar(PyObject* self, PyObject* args) {
+  HANDLE_TH_ERRORS
+  auto& self_ = reinterpret_cast<THPVariable*>(self)->cdata;
+  if (isFloatingType(self_.type().scalarType())) {
+    // we can't dispatch to toCLong here because we want to avoid ATen overflow checks;
+    // the python integral type (long in python2) can't overflow.
+    AutoGPU auto_gpu(self_);
+    Scalar localScalar = self_.get()->localScalar();
+    return THPUtils_packDoubleAsInt(localScalar.toDouble());
```

Comment:
These three lines could be:

```
return THPUtils_packDoubleAsInt(dispatch_to_CDouble(self_));
```

This has the advantage of releasing the GIL before doing a potentially blocking op.